components/availability/src/Availability.svelte

```diff
@@ -826,6 +826,19 @@
       slots: [],
     });
   }
+
+  // Track and dispatch active dates whenever they change
+  let canonicalDates: Date[] = [];
+  $: if (
+    JSON.stringify(canonicalDates) !==
+    JSON.stringify(days.map((day) => day.timestamp))
+  ) {
+    canonicalDates = days.map((day) => day.timestamp);
+    dispatchEvent("dateChange", {
+      dates: canonicalDates,
+    });
+  }
```

Comment:
This if statement just sounds like `days.length != 0`, because `canonicalDates` has to be `[]` and `days.map` doesn't change `days`'s length.
```suggestion
  $: if (days.length != 0) {
    dispatchEvent("dateChange", {
      dates: days.map((day) => day.timestamp),
    });
  }
```

Reply:
Hey @james10424 , `days` is a reactive variable; it updates elsewhere. What we're comparing here is the dates shown in days.

This string compare prevents multiple dispatches from firing to the parent app. If we commit your suggestion here, we'll get a `dateChanged` dispatch several times unnecessarily (the `days` array is modified, and therefore all `$: ` statements with days as a dependent fire, whenever someone hovers on a timeslot for example)

Reply:
Interesting to know, but don't you want it to fire when `days` is modified? So this won't get `dateChanged` dispatch because you converted it to a string? If you don't want to fire it when `days` is modified, then what's the point of `$:`?
What about the variable `canonicalDates`? `JSON.stringify(canonicalDates)` can totally be `JSON.stringify([])`.

Reply:
We set `canonicalDates` to something else on line 836.

Try firing this up locally and see what the proposed modifications do to the dispatch; you'll see it changes the frequency to unnecessarily often.